ras_party/controllers/account_controller.py

```diff
@@ -155,19 +156,31 @@ def change_respondent_enrolment_status(payload, session):
                 status=change_flag)
     respondent = query_respondent_by_party_uuid(respondent_id, session)
     if not respondent:
-        raise RasError("Respondent does not exist.", status=404)
+        raise RasError("Respondent does not exist", status=404)
 
     enrolment = query_enrolment_by_survey_business_respondent(respondent_id=respondent.id,
                                                               business_id=business_id,
                                                               survey_id=survey_id,
                                                               session=session)
     enrolment.status = change_flag
 
+    casegroup_ids = get_business_survey_casegroups(survey_id, business_id)
+
     category = 'DISABLE_RESPONDENT_ENROLMENT' if change_flag == 'DISABLED' else 'ENABLE_RESPONDENT_ENROLMENT'
     description = "Disable respondent enrolment" if change_flag == 'DISABLED' else 'Enable respondent enrolment'
-    for case in get_cases_for_collection_exercise(survey_id, business_id, respondent_id):
+    for case in get_cases_for_casegroups(casegroup_ids, respondent_id):
         post_case_event(case['id'], business_id, category=category, desc=description)
 
+    session.commit()
```

Comment:
What is this `session.commit()` for?

Reply:
Above here we've disabled an enrolment with `enrolment.status = change_flag`
Then on the line below this one we query the database to get the number of enabled enrolments. If we don't commit first that previous enrolment wont be disabled yet

Reply:
We could not commit and do a check for it there if the enrolment count was equal to '1' instead but I think it's better to actually disable the enrolment first

Reply:
The commit is fine, it just wasn't clear what it was committing. It's much clearer now the commit has been moved next to the database change.